src/views/Dashboard/Customer/Home/MyTest/StakingServiceTab.vue

```diff
@@ -90,78 +120,122 @@ export default {
-        stakeStatus: "Unstaked",
-        amount: 5000
-      }
-    ],
+    items: [],
     documents: null,
-    tabs: null
+    tabs: null,
+    showDialog: false,
+    requestId: ""
   }),
 
+  computed: {
+    ...mapState({
+      api: (state) => state.substrate.api,
+      pair: (state) => state.substrate.wallet,
+      web3: (state) => state.metamask.web3
+    })
+  },
+
+  async mounted () {
+    await this.fetchData ()
+  },
+
   methods: {
-    toRequestTest() {
+    ...mapMutations({
+      setCategory: "lab/SET_CATEGORY",
+      setStakingService: "lab/SET_STAKING_SERVICE"
+    }),
+
+    async fetchData () {
+      this.items = dataStaking.data // TO REMOVE when the indexer is working
+      await getServiceRequestByCustomer(this.pair.address)
+      
+    },
+
+    async getServiceRequest() {
+      // TODO : get service request
+    },
+
+    setAmount(amount) {
+      const formatedAmount = this.web3.utils.fromWei(String(amount), "ether")
+      return formatedAmount
+    },
+
+    setRemainingStakingDate(date) {
+      let day = new Date(parseInt(date)).getDate() + 6 - new Date().getDate()
+      let hours = new Date(parseInt(date)).getHours() + 23 - new Date().getHours()
+      let minutes = new Date(parseInt(date)).getMinutes() + 59 - new Date().getMinutes()
+
+      if (minutes > 60 ) {
+        hours-=1
+        minutes-=60
+      }
+
+      return `${day}D:${hours}H:${minutes}M`      
+    },
+
+    async toRequestTest() {
+      this.setStakingService(this.service)
+      this.$router.push({ 
+        name: "customer-request-test-service",
+        params: { flag: "staking" }
+      })
+
+      const country = this.service.country
+      const region = this.service.state
+      const city = this.service.city
+      const category = this.service.category
+      const flow = "StakingRequestService"
+
+      this.setCategory(category)
+
+      await this.$store.dispatch("lab/setCountryRegionCity", {country, region, city})
+      await this.$store.dispatch("lab/getServicesByCategory", {category, flow})
+      await this.$store.dispatch("rating/getRate")
+
       this.$router.push({ name: "customer-request-test-service"})
     },
 
+    getUnstakingDialog(id) {
+      this.showDialog = true
+      this.requestId = id
+    },
+
     setButtonBackground(status) {
       const colors = Object.freeze({
-        "STAKED": "#F60689",
-        "SERVICE AVAILABLE": "#5640A5",
-        "REQUEST TEST": "#4CBB65",
-        "WAITING FOR UNSTAKE": "#FAAD15",
+        "OPEN": "#F60689",
+        "CLAIMED": "#5640A5",
+        "PROCESS": "#4CBB65",
+        "WAITINGFORUNSTAKED": "#FAAD15",
         "UNSTAKED": "#E32319"
       })
 
       return { color: colors[status.toUpperCase()] }
+    },
+
+    closingDialog() {
+      console.log("close di sini")
+      this.showDialog = false
+    },
+
+    getStatusName(status) {
+      for (const key in stakingStatus) {
+        if (key === status.toUpperCase()) {
+          return stakingStatus[key]
```

Comment:
can we make the if statement shorter and reduce the number of lines? like this
`if (key === status.toUpperCase()) return stakingStatus[key]`

> hi, @bukandicki  where did you get the reference from? 😁

here:

- https://www.digitalocean.com/community/posts/5-tips-to-write-better-conditionals-in-javascript
- https://dev.to/sandricoprovo/how-to-write-cleaner-if-statements-in-javascript-2c0m